DiskSetup.sh

```diff
@@ -222,7 +228,7 @@ do
 done
 
 # See if our carve-target is an NVMe
-if [[ ${CHROOTDEV} =~ /dev/nvme ]]
+if [[ ${CHROOTDEV} =~ /dev/nvme0 ]]
```

Comment:
It looks vaguely odd to match on `nvme0n1` in one place and `nvme0` in another. Any particular reason to handle the comparison differently in each place?

Reply:
Previously, there's been no first NVMe conditional b/c spel wasn't potentially using NVMe at all. When I was adding the logic for XFS, the `pvcreate` had proven necessary in the "bare" use-case (i.e., "hand-testing" of the scripts where the secondary EBS was being re-used multiple times and for generating "recovery AMIs").

- The first `nvme` reference is focused on the root disk _specifically_ (to avoid the `pvcreate` on a `pivot_root`). The (avoided) `pvcreate` had been added due to issues encountered when using AMIgen7 bare (my primary case and the case used when creating "recovery AMIs"). 
- Second `nvme` reference is differentiating based on _any_ NVMe-type (mostly added to facilitate the "creating a recovery AMI" use-case, as noted above).

Reply:
Ok, but the second `nvme` reference is now to `nvme0`, which represents the root volume, right? If it were a second volume, it would be `nvme1`, I think.

Reply:
Negative:
* `/dev/nvme0`: first NVMe bus
* `/dev/nvme0n1`: first device on first NVMe bus
* `/dev/nvme0n1p1`: first partition on first device on first NVMe bus
Fun times. Notionally, the second one _c/should_ be simply `/dev/nvme` ...but AWS doesn't present devices on a second NVMe bus, so `/dev/nvme0` _should_ be safe


In general, NVMe is going to be a pain in the dick in the future: its relative naming combined with "the device that answers earliest gets the lowest dev-num" algorithm can be a righteous ass-bite (e.g., trying to account for two secondary EBSes in a CFn is a crap-shoot). Fortunately, `/dev/nvme0n1` is stable as the probing of the rest of the bus doesn't happen until after GRUB2 has already assigned the boot-dev to `/dev/nvme0n1`.

Reply:
Just realized, that was an accidental change. Prior statement still stands, but, going to put the second one back to what it was before all this.